packages/datatrak-web/src/features/Survey/useValidationResolver.ts

```diff
@@ -0,0 +1,106 @@
+import { useCallback } from 'react';
+import * as yup from 'yup';
+import { getAllSurveyComponents, useSurveyForm } from '.';
+import { QuestionType } from '@tupaia/types';
+
+export const useValidationResolver = () => {
+  const { visibleScreens, activeScreen, isLast, formData } = useSurveyForm();
+  const allSurveyComponents = getAllSurveyComponents(visibleScreens);
+
+  const componentsToValidate = isLast ? allSurveyComponents : activeScreen;
+
+  const validationSchema =
```

Comment:
Again I would find it much more readable if this function was split up a bit. I would split out the bit that turns the componentsToValidate into a validation schema
eg. 
const validationSchema = makeValidationSchema(componentsToValidate);